### PR TITLE
Move Dashboard's back-end logic to Lifecycle

### PR DESCRIPTION
**Pull request by iszulcdeepsense**
These changes has been extracted from https://github.com/TheRacetrack/racetrack/pull/232 to make it less enormous.
It contains all minor changes that had to be made to rewrite Dashboard, but not revolving around front-end stuff.

### CODE REVIEW

File: lifecycle/lifecycle/job/portfolio.py [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+            reasons.append('Job seems to be unused as it was never called.')
+        else:
+            last_call_days_ago = days_ago(job.last_call_time)
+            if last_call_days_ago > 1:
+                score -= 10 * _interpolate(last_call_days_ago, 0, 30)
+                reasons.append(f'Job hasn\'t been called for {last_call_days_ago:.1f} days.')
+
+    newer_versions = _count_job_newer_versions(job, all_jobs)
+    if newer_versions > 0:
+        score -= 1 * newer_versions
```
Comment (LookACastle) [inline lifecycle/lifecycle/job/portfolio.py:50]: This should be interpolated as well. You shouldn't remove a legacy job (that might be legacy for a reason) with a higher priority than a lost job if it's legacy enough.

Also, you should probably be able to escape this scoring as a job somehow. It's all but guaranteed that someone somewhere will be running a mission critical system that's 10 years out of date, and that shouldn't lead to it getting axed.

**Review by LookACastle — CHANGES_REQUESTED** [review]

File: racetrack_commons/racetrack_commons/auth/__init__.py
[the remark this answers is not in the document]
Comment (iszulcdeepsense) [reply racetrack_commons/racetrack_commons/auth/__init__.py:1]: It was never intended as renaming. I created this file, but for some reason git / github matched this file as a result of  renaming

File: lifecycle/lifecycle/job/ansi.py [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import re
+
+
+def remove_ansi_sequences(content: str) -> str:
+    """Remove ANSI escape codes controlling font colors"""
+    return re.sub(r'\x1b\[\d+(;\d+)?m', '', content)
```
[the remark this answers is not in the document]
Comment (LookACastle) [reply lifecycle/lifecycle/job/ansi.py:6]: `strip_ansi_colors` is good too.

**Review by LookACastle — APPROVED** [review]
All my feedback seems to be reacted to and solved, LGTM

**Pull request merged**